File: ycs-golang/core/content_factory_accessor.go
// ------------------------------------------------------------------------------
//  <copyright company="Microsoft Corporation">
//      Copyright (c) Microsoft Corporation.  All rights reserved.
//  </copyright>
// ------------------------------------------------------------------------------

package core

import (
	"github.com/chenrensong/ygo/contracts"
)

// ContentFactoryAccessor provides access to the content factory
type ContentFactoryAccessor struct{}

var factory contracts.IContentFactory

// SetFactory sets the content factory
func SetFactory(contentFactory contracts.IContentFactory) {
	factory = contentFactory
}

// GetFactory gets the content factory
func GetFactory() contracts.IContentFactory {
	return factory
}

File: src/Ycs/Core/ContentFactoryAccessor.cs
// ------------------------------------------------------------------------------
//  <copyright company="Microsoft Corporation">
//      Copyright (c) Microsoft Corporation.  All rights reserved.
//  </copyright>
// ------------------------------------------------------------------------------

using Ycs.Contracts;

namespace Ycs.Core
{
    /// <summary>
    /// Global accessor for ContentFactory to avoid circular dependencies
    /// This class is initialized by YcsBootstrap
    /// </summary>
    internal static class ContentFactoryAccessor
    {
        private static IContentFactory _factory;

        /// <summary>
        /// Sets the ContentFactory instance (called by YcsBootstrap)
        /// </summary>
        internal static void SetFactory(IContentFactory factory)
        {
            _factory = factory;
        }

        /// <summary>
        /// Gets the ContentFactory instance
        /// </summary>
        internal static IContentFactory Factory
        {
            get
            {
                return _factory ?? throw new System.InvalidOperationException("ContentFactory not initialized. Call YcsBootstrap.Initialize() first.");
            }
        }
    }
}